infrastructure/zk/src/database.ts

```diff
@@ -1,101 +1,199 @@
+    await utils.exec(`pg_isready -d "${dalDb}"`);
 }
 
-export async function checkSqlxData() {
-    process.chdir('core/lib/dal');
-    await utils.spawn('cargo sqlx prepare --check -- --tests');
+export async function wait(opts: any, tries: number = 4) {
+    let dals = getDals(opts);
+    for (const dalDb of dals.values()) {
+        await waitForDal(dalDb, tries);
+    }
+}
+
+async function checkSqlxDataForDal(dalPath: string, dalDb: string) {
+    process.chdir(dalPath);
+    await utils.spawn(`cargo sqlx prepare --check -- --tests`);
     process.chdir(process.env.ZKSYNC_HOME as string);
 }
 
+export async function checkSqlxData(opts: any) {
+    let dals = getDals(opts);
+    for (const [dalPath, dalDb] of dals.entries()) {
+        await checkSqlxDataForDal(dalPath, dalDb);
+    }
+}
+
 export const command = new Command('db').description('database management');
 
-command.command('drop').description('drop the database').action(drop);
-command.command('migrate').description('run migrations').action(migrate);
-command.command('new-migration <name>').description('generate a new migration').action(generateMigration);
-command.command('setup').description('initialize the database and perform migrations').action(setup);
-command.command('wait').description('wait for database to get ready for interaction').action(wait);
-command.command('reset').description('reinitialize the database').action(reset);
-command.command('reset-test').description('reinitialize the database for test').action(resetTest);
-command.command('check-sqlx-data').description('check sqlx-data.json is up to date').action(checkSqlxData);
+command.command('drop').description('drop the database').option('-p, --prover').option('-s, --server').action(drop);
+command.command('migrate').description('run migrations').option('-p, --prover').option('-s, --server').action(migrate);
+command
+    .command('new-migration <name>')
+    .description('generate a new migration')
+    .option('-p, --prover')
+    .option('-s, --server')
+    .action(generateMigration);
+command
+    .command('setup')
+    .description('initialize the database and perform migrations')
+    .option('-p, --prover')
+    .option('-s, --server')
+    .action(setup);
+command
+    .command('wait')
+    .description('wait for database to get ready for interaction')
+    .option('-p, --prover')
+    .option('-s, --server')
+    .action(wait);
+command
+    .command('reset')
+    .description('reinitialize the database')
+    .option('-p, --prover')
+    .option('-s, --server')
+    .action(reset);
+command
+    .command('reset-test')
+    .description('reinitialize the database for test')
+    .option('-p, --prover')
+    .option('-s, --server')
+    .action(resetTest);
+command
+    .command('check-sqlx-data')
+    .description('check sqlx-data.json is up to date')
+    .option('-p, --prover')
+    .option('-s, --server')
+    .action(checkSqlxData);
```

Comment:
I gave it a little thought and I agree with this interface - having separate, but consistent parameters to specify if you want server or prover